### DISCUSSION TITLE
LuaSnip problem

### DISCUSSION THREAD

**Discussion by gilfuser** [davidgranstrom/scnvim#198] [Q&A] [answered]
Hi there!
I use SCNvim for a while, but I had an outdated version and recently I'm catching up with the new stuff. Very cool stuff!

I'm trying to add LuaSnip to the SCNvim setup following [this post](https://madskjeldgaard.dk/posts/neovim-as-sc-ide/#snippets-luasnip) and I'm changing the init.vim for the init.lua.

I put this in the end of the luasnip.lua config file I copy pasted [from here](https://github.com/L3MON4D3/LuaSnip/blob/master/Examples/snippets.lua#L190):
`ls.snippets.supercollider = require'scnvim/utils'.get_snippets()`

I did the `:SCNvimGenerateAssets`  which generate the assets here:
```
Generated tags file: /home/skmecs/.cache/nvim/scnvim/tags
Generated syntax file: /home/skmecs/.cache/nvim/scnvim/classes.vim
Generated snippets file: /home/skmecs/.cache/nvim/scnvim/scnvim_snippets.lua
```
I checked and they are really there, with their content and all.

but then, when I save the buffer and `: source %` or open a buffer I get this error.
```sc
Error detected while processing /home/skmecs/.config/nvim/lua/plugins/luasnip.lua:
E5113: Error while calling lua chunk: /home/skmecs/.config/nvim/lua/plugins/luasnip.lua:564: attempt to index field 'snippets' (a nil value)
stack traceback:
        /home/skmecs/.config/nvim/lua/plugins/luasnip.lua:564: in main chunk
```

In my scnvim.lua config file I put this:
`vim.g.scnvim_snippet_format = 'luasnip'`

If I try `SinOsc.ar` `<C-k>` nothing happens, as spected.
My brand new init.lua looks like this:
```lua
vim.cmd [[
    syntax enable
    filetype plugin on
]]
local o = vim.o
local g = vim.g
local set = vim.opt

require "paq" {
    "savq/paq-nvim";                  -- Let Paq manage itself

    "neovim/nvim-lspconfig";          -- Mind the semi-colons
    "hrsh7th/nvim-cmp";
    'hrsh7th/cmp-nvim-lsp';
    'hrsh7th/cmp-buffer';
    'hrsh7th/cmp-path';
    'hrsh7th/cmp-cmdline';
    'hrsh7th/nvim-cmp';

    'L3MON4D3/LuaSnip';
    'saadparwaiz1/cmp_luasnip';
    'windwp/nvim-autopairs';
    'numToStr/Comment.nvim';
    'karb94/neoscroll.nvim';

    'nvim-treesitter/nvim-treesitter';
    'JoosepAlviste/nvim-ts-context-commentstring';
    'nvim-treesitter/nvim-treesitter-context';
    'p00f/nvim-ts-rainbow';
    'nvim-treesitter/nvim-treesitter-refactor';
    'nvim-treesitter/playground';
    { 'junegunn/fzf', run = './install --bin', };
    'kyazdani42/nvim-web-devicons';
    -- {'ibhagwan/fzf-lua', branch = 'main' };
    'vijaymarupudi/nvim-fzf';
    'davidgranstrom/scnvim';
    'madskjeldgaard/fzf-sc';
    {"lervag/vimtex", opt=true};      -- Use braces when passing options

    "ellisonleao/gruvbox.nvim";
    "tidalcycles/vim-tidal";
    'zhou13/vim-easyescape';
    'nvim-lualine/lualine.nvim';
}

g.mapleader = 'ö'
g.maplocaleader = ','

g.python2_host_prog = "/usr/bin/python2"
g.python3_host_prog = "/usr/bin/python3"

require ('skmecs.settings')
require ('skmecs.themes')
require ('plugins.easy-escape')
require ('plugins.nvim-cpm')
require ('plugins.autopairs')
require ('plugins.comment')
require('plugins.neoscroll')
require('plugins.lualine')
require('plugins.treesitter')
require('plugins.scnvim')
require('plugins.luasnip')

```

Everything but LuaSnip, playground and fzf-sc work fine. I'll make another Q for the latter.

Thank you in advance for any direction pointed.
If I could give any info that could help undertand what's happening, please ask.

bests,
Gil

**Comment by davidgranstrom** [comment]
Hi,

I spotted two things from the config and error message.

1) `vim.g.scnvim_snippet_format = 'luasnip'` is no longer a supported option, instead specify `snippet.engine.name` in the config like this:

```lua
snippet = {
  engine = {
    name = 'luasnip',
  },
},
```
 See this [page](https://github.com/davidgranstrom/scnvim/wiki/Configuration) for an overview off the default options, and check out the [reference](https://davidgranstrom.github.io/scnvim/modules/scnvim.config.html) for detailed descriptions.

2)

```lua
Error detected while processing /home/skmecs/.config/nvim/lua/plugins/luasnip.lua:
E5113: Error while calling lua chunk: /home/skmecs/.config/nvim/lua/plugins/luasnip.lua:564: attempt to index field 'snippets' (a nil value)
stack traceback:
        /home/skmecs/.config/nvim/lua/plugins/luasnip.lua:564: in main chunk
```

This error message is not related to scnvim but to `luasnip`. Possibly you are missing a `require` for luasnip at the beginning of your script?

```lua
local ls = require'luasnip'
ls.snippets.supercollider = require'scnvim/utils'.get_snippets()
```

**Reply by gilfuser** [reply]
`/home/skmecs/.config/nvim/lua/plugins/luasnip.lua` line `564` is exactly this line:
```lua
ls.snippets.supercollider = require'scnvim/utils'.get_snippets()
```
I tried it in the config file for scnvim and in the one for luasnip. With `local ls = require'luasnip'` at the beggining.

**Comment by davidgranstrom** [comment, marked as answer]
There seems to have been a breaking change in the LuaSnip API: https://github.com/L3MON4D3/LuaSnip/issues/81#issuecomment-1073301357

This works for me:
```lua
ls.add_snippets('supercollider', {
  s('sb', { t('s.boot;') }),
}, {
  key = 'supercollider'
})
```

And

```lua
ls.add_snippets('supercollider', require'scnvim.utils'.get_snippets())
```

..to use the generated snippets. Note that this file is pretty big so adding the snippets it will affect the overall startup time of nvim. If you plan to use the generated snippets you might consider loading them by using an autocmd instead.
